src/views/AllNotice.vue

```diff
@@ -0,0 +1,144 @@
+              <div style="text-align:right">
+                  <span style="color:blue"><small>By: {{not.user}}</small></span>
+              </div>
+              <div style="text-align:left">
+                  <span><strong>{{not.notices}}</strong></span>
+              </div>
+              <div style="text-align:right">
+                  <span style="color:blue"><small>{{not.timestamp}}</small></span>
+              </div>
+          </li>
+      </ul>
+     
+      <ul v-else>
+        <li>There is no notice at present right now</li>
+      </ul>
+        
+      </div>
+       
+  </section>
+</template>y
+
+
+<script>
+import firebase from 'firebase';
+export default {
+ 
+  name: "AllNotice",
+  data(){
+      return{
+          noticearr:[],
+          
+      }
+
+  },
+  methods:{
+      async fetchnotice(){
+          var db=firebase.firestore();
+          db.collection('noticeboard').orderBy('timestamp','desc').onSnapshot((querySnapshot)=>{
+              let allnotice=[];
+              querySnapshot.forEach(doc=>{
+                  allnotice.push(doc.data())
+              })
+            
+              this.noticearr=allnotice;
+          })
+      }
+  },
+  created(){
+      this.fetchnotice();
+  },
+  beforeRouteEnter(to,from,next){
+      next(vm=>{
+          firebase.auth().onAuthStateChanged(user=>{
+              if(user){
+                  vm.$router.push('/allnotice');
+
+              }
+              else{
+                  alert('You need to Sign-in to see the Noticeboard');
+                  vm.$router.push('/login');
+              }
+          })
+      })
+  }
+};
+</script>
+
+<style scoped>
+.body{
+    background: linear-gradient(to bottom right, white, #bcbdc4);
+    padding: 15px;
+    min-height:400px;
+
+}
+.body1{
+    opacity: 0.1;
+}
+.heading{
+    background-image: url('../assets/walpaper.jpg');
+    background-size:40rem;
+    margin:auto;
+    width:90%;
+    max-width:40rem;
+    padding:15px;
+    border-radius:10px;
+    color:whitesmoke;
+    text-align: center;
+
+
+}
+p{
+  font-size: large;  
+}
+ul{
+    margin: 0;
+    padding: 0;
+    list-style: none;
+}
+li{
+    box-shadow: 0 2px 8px #1e0c42;
+    margin: 1rem auto;
+    border-radius: 10px;
+    padding: 1rem;
+    text-align: center;
+    width: 90%;
+    max-width: 40rem;
+    background-color:whitesmoke;
+    font-family: 'Times New Roman', Times, serif;
+    font-size: larger;
+    
+    
+}
+button{
+    font: inherit;
+    cursor: pointer;
+    border: 1px solid #ff0077;
+    background-color: #ff0077;
+    color: white;
+    padding: 0.05rem 1rem;
+    box-shadow: 1px 1px 2px rgba(0, 0, 0, 0.26);
+}
+button:hover{
+    background-color: #ec3169;
+    border-color: #ec3169;
+    box-shadow: 1px 1px 4px rgba(0, 0, 0, 0.26);
+}
+.inp{
+    margin:1rem auto;
+    width:90%;
+    max-width:50rem;
+    padding:15px;
+    border-radius:10px;
+    color:black;
+    /* text-align: center; */
+}
```

Comment:
Why is this CSS required in this file?

Reply:
I changed input field to text-area but forgot to remove the input css. I will remove that